src/pip/_internal/operations/install/wheel.py

```diff
@@ -135,24 +136,23 @@ def message_about_scripts_not_on_PATH(scripts: Sequence[str]) -> Optional[str]:
         return None
 
     # Group scripts by the path they were installed in
-    grouped_by_dir: Dict[str, Set[str]] = collections.defaultdict(set)
+    grouped_by_dir: Dict[Path, Set[str]] = collections.defaultdict(set)
     for destfile in scripts:
-        parent_dir = os.path.dirname(destfile)
-        script_name = os.path.basename(destfile)
+        parent_dir = Path(destfile).parent.resolve()
+        script_name = Path(destfile).name
         grouped_by_dir[parent_dir].add(script_name)
 
     # We don't want to warn for directories that are on PATH.
     not_warn_dirs = [
-        os.path.normcase(i).rstrip(os.sep)
-        for i in os.environ.get("PATH", "").split(os.pathsep)
+        Path(i).resolve() for i in os.environ.get("PATH", "").split(os.pathsep)
```

Comment:
This requires all paths in `PATH` to exist, which is not necessarily true. So we need to catch FileNotFoundError and ignore those.

Reply:
according to the [docs](https://docs.python.org/3.11/library/pathlib.html#pathlib.Path.resolve) they do not have to exists